cloudvault_discovery/core/exploiter/credential_validator.py

```
@@ -0,0 +1,228 @@
+    def validate_credentials(self, credentials: List[Dict]) -> List[Dict]:
+        """Validate list of credentials"""
+        results = []
+        with concurrent.futures.ThreadPoolExecutor(max_workers=10) as executor:
+            futures = [executor.submit(self._validate_single_credential, cred) for cred in credentials]
+            for future in concurrent.futures.as_completed(futures, timeout=120):
+                try:
+                    result = future.result()
+                    if result and result.get('valid'):
+                        results.append(result)
+                        self.validated_creds.append(result)
+                except:
```

Ruff (0.14.3)

Do not use bare `except`

(E722)